scipy/sparse/linalg/isolve/iterative/BiCGREVCOM.f.src

```diff
@@ -319,7 +320,13 @@
 *****************
  6       CONTINUE
 *****************
-         ALPHA = RHO / <xdot>( N, WORK(1,PTLD), 1, WORK(1,Q), 1 )
+         TMPVAL = <xdot>( N, WORK(1,PTLD), 1, WORK(1,Q), 1 )
+         IF (TMPVAL.EQ.0) THEN
+*           Breakdown
+            INFO = -11
+            GO TO 20
+         ENDIF
+         ALPHA = RHO / TMPVAL
```

Comment:
Is abstracting the duplicated error handling logic to a function (or similar) in Fortran more trouble than it is worth?

And no better option than `GO TO`? I guess for error handling this is pretty common.

Reply:
I'd imagine it's more trouble. The goto usage is quite typical for old Fortran 77 code.

Reply:
I guess most languages also block goto from breaking out of functions if I'm not mistaken, to prevent complete chaos, so that's probably a second excuse not to abstract.